jtec-agent/src/main/java/edu/tum/sse/jtec/agent/Tracer.java

```diff
@@ -1,50 +1,83 @@
+            final Path testEventOutput = options.getOutputPath().resolve(String.format("%s_%d_test.log", getCurrentPid(), System.currentTimeMillis()));
             createFileAndEnclosingDir(testEventOutput);
-            customInstrumentationList.add(new TestEventInstrumentation(testEventOutput.toString()).attach(instrumentation));
+            customInstrumentationList.add(new TestEventInstrumentation(testEventOutput.toString()).attach(instrumentation, tempFolder));
         }
         if (options.shouldTraceSystemEvents()) {
-            Path sysEventOutput = options.getOutputPath().resolve(String.format("%s_%d_sys.log", getCurrentPid(), System.currentTimeMillis()));
+            final Path sysEventOutput = options.getOutputPath().resolve(String.format("%s_%d_sys.log", getCurrentPid(), System.currentTimeMillis()));
             createFileAndEnclosingDir(sysEventOutput);
-            customInstrumentationList.add(new SysEventInstrumentation(sysEventOutput.toString()).attach(instrumentation));
+            customInstrumentationList.add(new SysEventInstrumentation(sysEventOutput.toString()).attach(instrumentation, tempFolder));
         }
         if (options.shouldTraceCoverage()) {
-            Path covEventOutput = options.getOutputPath().resolve(String.format("%s_%d_cov.log", getCurrentPid(), System.currentTimeMillis()));
+            final Path covEventOutput = options.getOutputPath().resolve(String.format("%s_%d_cov.log", getCurrentPid(), System.currentTimeMillis()));
             customInstrumentationList.add(
                     new CoverageInstrumentation(
                             covEventOutput.toString(),
                             options.getCoverageLevel(),
                             options.getCoverageIncludes(),
                             options.getCoverageExcludes(),
                             options.shouldInstrumentCoverage()
-                    ).attach(instrumentation));
+                    ).attach(instrumentation, tempFolder));
+        }
+    }
+
+    /**
+     * Returns the directory that contains the agent or null if it can't be resolved.
+     */
+    public static Path getJTecInstallDirectory() {
+        try {
+            final URI jarFileUri = IOUtils.class.getProtectionDomain().getCodeSource().getLocation().toURI();
+            // we assume that the dist zip is extracted and the agent jar not moved
+            return Paths.get(jarFileUri).getParent().getParent().getParent();
+        } catch (final URISyntaxException e) {
+            throw new RuntimeException("Failed to obtain agent directory. This is a bug, please report it.", e);
+        }
+    }
+
+    private String getInstrumentationLocation() {
+        final Path jTecInstallDirectory = getJTecInstallDirectory();
+        final String implementationVersion = getClass().getPackage().getImplementationVersion();
+        if (jTecInstallDirectory.toAbsolutePath().toString().contains(".m2")) {
```

Comment:
This will not work with custom Maven repo locations (and non-maven projects). I think we should switch to a different approach for getting the JARs location.